conformance/conformance_python.py

```diff
@@ -56,14 +57,26 @@ def do_test(request):
   test_message = test_messages_proto3_pb2.TestAllTypes()
   response = conformance_pb2.ConformanceResponse()
   test_message = test_messages_proto3_pb2.TestAllTypes()
+  test_message_proto2 = test_messages_proto2_pb2.TestAllTypesProto2()
+  isProto3 = (request.message_type == "proto3")
+  isJson = (request.WhichOneof('payload') == 'json_payload')
 
   try:
     if request.WhichOneof('payload') == 'protobuf_payload':
-      try:
-        test_message.ParseFromString(request.protobuf_payload)
-      except message.DecodeError as e:
-        response.parse_error = str(e)
-        return response
+      if isProto3:
+        try:
+          test_message.ParseFromString(request.protobuf_payload)
```

Comment:
It should be possible to share most of the code between proto2 and proto3. Something like:

```
  test_message = <choose based on proto2/proto3>
  if request.WhichOneof('payload') == 'protobuf_payload':
    test_message.ParseFromString(request.protobuf_payload)
```